lib/nx_deflate.c

```diff
@@ -887,7 +887,8 @@ static void nx_compress_block_get_cpb(nx_streamp s, int fc)
 /* updates stream offsets and also sets the block final bit */
 static int  nx_compress_block_update_offsets(nx_streamp s, int fc)
 {
-	uint32_t copy_bytes, histbytes, overflow;
+	uint32_t copy_bytes, overflow;
+	volatile uint32_t histbytes;
```

Comment:
I didn't understand why `histbytes` gets volatile here.
Shouldn't `s->nxcmdp->cpb` and its fields actually be volatile?

Are you doing this in order to avoid applying volatile in code that does not require extra reads?

Reply:
@tuliom I applied `volatile` to all variables that are set using `get*` macros, which includes `histbytes` here. But on a second thought, it makes more sense to mark CPB as volatile instead because those will be addresses where the engine will write to.

And looks like it's already volatile:
https://github.com/libnxz/power-gzip/blob/828e9ee6354de3886f5bc33348469a914261dfd6/inc_nx/nxu.h#L358